src/newdb.cpp

```diff
@@ -1105,22 +1105,22 @@ static bool save_char(char_data *player, DBIndex::vnum_t loadroom, bool fromCopy
   /* Remove their worn equipment to inventory. */
   for (i = 0; i < NUM_WEARS; i++) {
     if (player->equipment[i])
-      char_eq[i] = unequip_char(player, i, FALSE);
+      char_eq[i] = unequip_char(player, i, FALSE, FALSE);
     else
       char_eq[i] = NULL;
   }
 
   /* Remove their cyberware to inventory. */
   for (temp = player->cyberware; temp; temp = next_obj) {
     next_obj = temp->next_content;
-    obj_from_cyberware(temp);
+    obj_from_cyberware(temp, FALSE);
     obj_to_char(temp, player);
   }
 
   /* Remove their bioware to inventory. */
   for (temp = player->bioware; temp; temp = next_obj) {
     next_obj = temp->next_content;
-    obj_from_bioware(temp);
+    obj_from_bioware(temp, FALSE);
```

Comment:
If there isn't another guaranteed `affect_total()` call after this, you must put one in right before the SQL string is composed (~1170), otherwise character stats will balloon every time they log out and back in.

Reply:
Doesn't the affect_total in load_char prevent stats from ballooning?

Reply:
No, because you're saving it with increased stats. Take the example of muscle aug 4 on a strength 1 character:

1. at function start, str = 5
2. after removal, str = 5 (previously, str = 1)
3. at SQL write, str = 5 (previously, str = 1)
4. at re-apply, str = 5 (previously, str = 1)
5. at function exit, str = 5

Then load that char from DB:

1. at function start, str = 5
2. at bioware apply, str = 9
3. at function exit, str = 9

That's my current understanding, although admittedly I haven't read the code in that area in a while. If I've gotten it wrong, let me know!

Reply:
It looks to me like we're saving real attributes, which we shouldn't be modifying anywhere. Still, this should be tested to confirm (and regrettably, I don't have a testing environment set up).

For some reason, it looks like we do save hacking and combat pool dice to the database, so those will be out of sync with their saved attributes, but affect_modify calculates those pool limits from scratch anyway.